examples/07-Train-an-xgboost-model-using-the-Merlin-Models-API.ipynb

```diff
@@ -0,0 +1,346 @@
+{
```

Comment:
<p>It doesn't look like we're passing this directly to the model. The schema passed in still has both target columns. However, the logic in the code currently picks the target column based on the objective. e.g. objective="binary:logistic" -&gt; will look for targets in schema that are tagged with Tags.BINARY_CLASSIFICATION. And in this case rating has tags (Tags.REGRESSION, Tags.TARGET), and rating_binary has tags (Tags.BINARY_CLASSIFICATION, Tags.TARGET). So if the objective were switched from "binary:logistic" to "reg:logistic" then the target will be switched from rating_binary to rating.</p><p> </p><p>OBJECTIVES = {</p><p>&nbsp;&nbsp;"binary:logistic": Tags.BINARY_CLASSIFICATION,</p><p>&nbsp;&nbsp;"reg:logistic": Tags.REGRESSION,</p><p>&nbsp;&nbsp;"reg:squarederror": Tags.REGRESSION,</p><p>&nbsp;&nbsp;"rank:pairwise": Tags.TARGET,</p><p>&nbsp;&nbsp;"rank:ndcg": Tags.TARGET,</p><p>&nbsp;&nbsp;"rank:map": Tags.TARGET,</p><p>}</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/522/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>If this feels like too much magic. We could simplify the default (when no target columns are specfified) to rely only on the TARGET tag (e.g. all columns in schema with TARGET tag become the target for the model) and remove the extra complexity involving the objective and other tags. </p>

Reply:
this is some amazing sleuthing there Oliver! 🙂 outstanding

I had an earlier version of the example where I was passing in `target_columns` and then I must have omitted it here and it still worked. 

I really like how this automatic selection of the target column provides a glimpse into how the Merlin Models API works, and what value it can add, thus I am adding a bit of prose to explain how it actually works, based on what you shared above.

Reply:
This looks great Radek 🙂